lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1957,17 +1962,17 @@ SmallVector<int> NvidiaMmaEncodingAttr::getMMAv1ShapePerWarp(int opIdx) const {
 int NvidiaMmaEncodingAttr::getMMAv1Vec(int opIdx) const {
   return 2 * getMMAv1Rep(opIdx)[opIdx];
 }
-SmallVector<int64_t> NvidiaMmaEncodingAttr::getMMAv2Rep(ArrayRef<int64_t> shape,
+SmallVector<int64_t> NvidiaMmaEncodingAttr::getMMAv2OrV3Rep(ArrayRef<int64_t> shape,
                                                         int bitwidth,
                                                         int opIdx) const {
+  assert(isAmpere() || isHopper());
   auto rank = shape.size();
   auto warpsPerCTA = getWarpsPerCTA();
   SmallVector<int> shapePerWarp = {1, 16, 8, 4 * 64 / bitwidth};
```

Comment:
What about the `n` value here in Hopper?

Reply:
Ok, after having a thought about this, I guess that in all these places that depend on `instrN`, it's fine to leave `n = 8`, as the `n` pattern repeats with period `8` for the different tile sizes.

It would be good to leave a note somewhere in the code and refer to that note in all the places where we use this "trick" (here, in the SharedEncodingAttr, in the shared to dot operand mma, etc).

Reply:
For Hopper actually `opIdx` should always be 0 zero it doesn't support operand B in registers, and so we shouldn't ever use the n value.

I can add an assert here for `opIdx == 0`

Reply:
right! Adding an assert would be great.